src/emulator/storage/cloudFunctions.ts

```diff
@@ -69,6 +85,22 @@ export class StorageCloudFunctions {
       }, // bucket
       data: objectMetadataPayload,
     });
+    /** Cloud Event */
+    const ceAction = STORAGE_V2_ACTION_MAP[action];
+    const data = { ...objectMetadataPayload };
+    delete (data as any).acl; // remove this field from cloud events
+    const ce = {
+      specVersion: 1,
+      type: `google.cloud.storage.object.v1.${ceAction}`,
+      source: `//storage.googleapis.com/projects/_/buckets/${objectMetadataPayload.bucket}/objects/${objectMetadataPayload.name}`,
+      data,
+    };
+    const cloudEvent: string = JSON.stringify({
+      eventType: `google.cloud.storage.object.v1.${ceAction}`, // need eventType for multicast triggerId
+      data: ce,
+    });
```

Comment:
Huh I'm surprised that this works since I expected the body of the cloudevent http call to look like:

```js
{
  specVersion: 1,
  type: "google.cloud.storage.object.v1..."
  source: "..."
  data: { ... }
}
```
vs

```js
{
  data: {
    specVersion: 1,
    type: "google.cloud.storage.object.v1..."
    source: "..."
    data: { ... }
  }
}
```

Something must be unboxing the nested `data` property before sending it off to the function? Can you help me step through how this works?